src/wat-writer.cc

```diff
@@ -464,6 +490,8 @@ void WatWriter::WriteConst(const Const* const_) {
 }
 
 void WatWriter::WriteExpr(const Expr* expr) {
+  if (s_trace)
```

Comment:
I'm not a huge fan of keeping debug tracing in, since I find it to clutter up the code. I'd be OK with it though if we could minimize a bit... something like this maybe?
```
TRACE_ENTER("(%s)", GetExprTypeName(*expr))
TRACE_EXIT(...)
```

I think you could write the macro like:
```
#define TRACE_ENTER(fmt, ...) \
  if (s_trace) fprintf(stderr, "-> %s" fmt "\n", __func__, __VA_ARGS__)
```